kernel/src/vm/vmar/mod.rs

```diff
@@ -1384,4 +1385,143 @@ mod test {
             (_, None)
         ));
     }
+
+    /// Helper function to create a dummy `IoMem` for testing.
+    #[expect(unsafe_code)]
+    fn create_dummy_iomem(paddr: usize, size: usize) -> IoMem {
+        // SAFETY: This is for testing only, we use a safe physical address range
+        unsafe { IoMem::new(paddr..paddr + size, PageFlags::RW, CachePolicy::Uncacheable) }
+    }
+
+    #[ktest]
+    fn test_cow_copy_pt_iomem() {
+        let vm_space = VmSpace::new();
+        let map_range = PAGE_SIZE..(PAGE_SIZE * 2);
+        let cow_range = 0..PAGE_SIZE * 512 * 512;
+        let page_property = PageProperty::new_user(PageFlags::RW, CachePolicy::Uncacheable);
+        let preempt_guard = disable_preempt();
+
+        // Create and map IoMem instead of a frame.
```

Comment:
Be consistent to existing code:
```suggestion
        // Creates and maps an `IoMem` instead of a frame.
```